src/main/java/quickcarpet/feature/dispenser/PickupBucketablesBehavior.java

```diff
@@ -42,7 +49,27 @@ private ItemStack getBucketableAnimals(World world, BlockPos pos) {
         if (list.size() >= 1) {
             Entity first = list.get(0);
             first.remove(Entity.RemovalReason.DISCARDED);
-            return ((Bucketable) first).getBucketItem();
+
+            ItemStack stack = ((Bucketable) first).getBucketItem();
+
+            if(first instanceof AxolotlEntity){
```

Comment:
Why aren't you using `((Bucketable) first).copyDataToStack(stack)` here, but copy two implementations?